# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLMediaElement.cpp
# Commit: f42e3e5a600d
# Full Hash: f42e3e5a600d50db4c482ef83b8d5ee1a6eec5a7
# Author: alwu <alwu@mozilla.com>
# Date: 2020-07-24 09:32:06
# Regressor Bug: 1653390
# File Overlap Count: 1
# Description:
#   Bug 1653390 - part2: allow calling 'UpdateMediaAudibleState()' before the listener starts. r=bryce
#   
#   We already have a check to prevent notifying audible state before the listener starts [1], so we can remove the assertion and call it anytime in order to save the audible state correctly.
#   
#   [1] https://searchfox.org/mozilla-central/rev/dcd9c2d2bc19d96d487825eb70c2333a4d60994e/dom/html/HTMLMediaElement.cpp#478-486
# ==============================================================================

diff -r e2dc8fd47bc2 -r f42e3e5a600d dom/html/HTMLMediaElement.cpp
--- a/dom/html/HTMLMediaElement.cpp	Thu Jul 23 21:32:43 2020 +0000
+++ b/dom/html/HTMLMediaElement.cpp	Thu Jul 23 21:32:51 2020 +0000
@@ -466,9 +466,10 @@
     }
   }
 
+  // This method can be called before the listener starts, which would cache
+  // the audible state and update after the listener starts.
   void UpdateMediaAudibleState(bool aIsOwnerAudible) {
     MOZ_ASSERT(NS_IsMainThread());
-    MOZ_ASSERT(IsStarted());
     if (mIsOwnerAudible == aIsOwnerAudible) {
       return;
     }
@@ -7393,7 +7394,7 @@
   if (mAudioChannelWrapper) {
     mAudioChannelWrapper->NotifyAudioPlaybackChanged(aReason);
   }
-  if (mMediaControlKeyListener && mMediaControlKeyListener->IsStarted()) {
+  if (mMediaControlKeyListener) {
     mMediaControlKeyListener->UpdateMediaAudibleState(IsAudible());
   }
   // only request wake lock for audible media.
